lib/mayaUsd/render/vp2RenderDelegate/material.h

```diff
@@ -84,13 +84,17 @@ class HdVP2Material final : public HdMaterial
     const TfTokenVector& GetRequiredPrimvars() const { return _requiredPrimvars; }
 
 private:
+    void _ApplyVP2Fixes(HdMaterialNetwork& outNet, const HdMaterialNetwork& inNet);
     MHWRender::MShaderInstance* _CreateShaderInstance(const HdMaterialNetwork& mat);
     void                        _UpdateShaderInstance(const HdMaterialNetwork& mat);
     const HdVP2TextureInfo&     _AcquireTexture(const std::string& path);
 
     HdVP2RenderDelegate* const
         _renderDelegate; //!< VP2 render delegate for which this material was created
 
+    std::unordered_map<SdfPath, SdfPath, SdfPath::Hash>
```

Comment:
How long is this container needed during the lifetime of the material? In other words, does it have to be a member?

Reply:
It is needed during the whole lifetime of the material. Each material needs the mapping info to find the converted node names which are used to get fragment parameters.